core/io/resource.cpp

```diff
@@ -181,8 +181,39 @@ Error Resource::copy_from(const Ref<Resource> &p_resource) {
 			continue; //do not change path
 		}
 
-		set(E.name, p_resource->get(E.name));
+		Variant value = p_resource->get(E.name);
+
+		// The local-to-scene subresource instance is preserved, thus maintaining the previous sharing relationship.
+		// This is mainly used when the sub-scene root is reset in the main scene.
+		if (p_keep_local_to_scene_resource_instance) {
+			Ref<Resource> sub_res_of_from = value;
+			if (sub_res_of_from.is_valid() && sub_res_of_from->is_local_to_scene()) {
+				// Find the shared copy of the source resource.
+				HashMap<Ref<Resource>, Ref<Resource>>::Iterator R = p_remap_cache.find(sub_res_of_from);
+				if (R) {
+					value = R->value;
+				} else {
+					Ref<Resource> sub_res = get(E.name);
+					if (sub_res.is_valid() && sub_res->is_local_to_scene()) {
+						// Simply copy the data from the source resource to update the resource that was previously set.
+						sub_res->copy_from(sub_res_of_from, p_remap_cache, true);
+						p_remap_cache[sub_res_of_from] = sub_res;
+						value = sub_res;
+					} else { // A copy of the source resource is required to overwrite the previous one
```

Comment:
```suggestion
					} else { // A copy of the source resource is required to overwrite the previous one.
```
Code style nitpick, same elsewhere